keras_contrib/layers/normalization.py

```diff
@@ -1,3 +1,235 @@
+        self.running_std = self.add_weight(shape, initializer='one',
+                                           name='{}_running_std'.format(self.name),
+                                           trainable=False)
+
+        self.r_max = K.variable(np.ones((1,)), name='{}_r_max'.format(self.name))
+
+        self.d_max = K.variable(np.zeros((1,)), name='{}_d_max'.format(self.name))
+
+        self.t = K.variable(np.zeros((1,)), name='{}_t'.format(self.name))
+
+        if self.initial_weights is not None:
+            self.set_weights(self.initial_weights)
+            del self.initial_weights
+        self.built = True
+
+    def call(self, x, mask=None):
+        if self.mode == 0 or self.mode == 2:
+            assert self.built, 'Layer must be built before being called'
+            input_shape = K.int_shape(x)
+
+            reduction_axes = list(range(len(input_shape)))
+            del reduction_axes[self.axis]
+            broadcast_shape = [1] * len(input_shape)
+            broadcast_shape[self.axis] = input_shape[self.axis]
+
+            if K.backend() == 'tf':
+                mean_batch, var = K.KTF.nn.moments(x, reduction_axes, shift=None, name=None, keep_dims=False)
+                std_batch = (K.sqrt(var + self.epsilon))
+            else:
+                mean_batch = K.mean(x, axis=reduction_axes)
+                std_batch = K.sqrt(K.var(x, axis=reduction_axes) + self.epsilon)
+
+            r_max_val = K.get_value(self.r_max)
+            r = std_batch / self.running_std
+            r = K.stop_gradient(K.clip(r, 1 / r_max_val, r_max_val))
+
+            d_max_val = K.get_value(self.d_max)
+            d = (mean_batch - self.running_mean) / self.running_mean
+            d = K.stop_gradient(K.clip(d, -d_max_val, d_max_val))
+
+            if sorted(reduction_axes) == range(K.ndim(x))[:-1]:
+                x_normed_batch = (x - mean_batch) / std_batch
+                x_normed = (x_normed_batch * r + d) * self.gamma + self.beta
+            else:
+                # need broadcasting
+                broadcast_mean = K.reshape(mean_batch, broadcast_shape)
+                broadcast_std = K.reshape(std_batch, broadcast_shape)
+                broadcast_r = K.reshape(r, broadcast_shape)
+                broadcast_d = K.reshape(d, broadcast_shape)
+                broadcast_beta = K.reshape(self.beta, broadcast_shape)
+                broadcast_gamma = K.reshape(self.gamma, broadcast_shape)
+
+                x_normed_batch = (x - broadcast_mean) / broadcast_std
+                x_normed = (x_normed_batch * broadcast_r + broadcast_d) * broadcast_gamma + broadcast_beta
+
+            # explicit update to moving mean and standard deviation
+            self.add_update([K.moving_average_update(self.running_mean, mean_batch, self.momentum),
+                             K.moving_average_update(self.running_std, std_batch ** 2, self.momentum)], x)
+
+            # update r_max and d_max
+            t_val = K.get_value(self.t)
```

Comment:
It is. The user can specify r_max_value and d_max_value. That will change the scale. If you mean the rate at which the value changes (the algorithm), I tried out 3-4 different algorithms that change the rate, this one is by far the smoothest and fastest. 

For reference: I have tried linear, squared, general population growth, derivative based approaches and finally biological population growth formula. The last one performs the best. 

The author of the paper simply states "For Batch Renorm, we used rmax = 1, dmax = 0 (i.e. simply batchnorm) for the first 5000 training steps, after which these were gradually relaxed to reach rmax = 3 at 40k steps, and dmax = 5 at 25k steps.". Since we cannot access "iteration count" or "epoch" count inside a layer, I had to resort to using my own time counter to perform such a scaling. 

I do not think there is any "simple" way of allowing the user to adjust the rate, and a callback approach is not my idea of good design.